pytorch_lightning/loggers/neptune.py

```diff
@@ -46,247 +93,469 @@ class NeptuneLogger(LightningLoggerBase):
 
         Example::
 
-            self.logger.experiment.some_neptune_function()
+            class LitModel(LightningModule):
+                def training_step(self, batch, batch_idx):
+                    # log metrics
+                    acc = ...
+                    self.logger.experiment["train/acc"].log(acc)
 
+                    # log images
+                    img = ...
+                    self.logger.experiment["train/misclassified_images"].log(File.as_image(img))
         """
+        return self.run
 
-        # Note that even though we initialize self._experiment in __init__,
-        # it may still end up being None after being pickled and un-pickled
-        if self._experiment is None:
-            self._experiment = self._create_or_get_experiment()
-
-        return self._experiment
+    @property
+    def run(self) -> Run:
+        if self._run_instance is None:
+            try:
+                self._run_instance = neptune.init(
+                    project=self._project,
+                    api_token=self._api_key,
+                    name=self._name,
+                    **self._neptune_run_kwargs,
+                )
+            except NeptuneLegacyProjectException as e:
+                raise TypeError(
+                    f"""Project {self._project} has not been migrated to the new structure.
+                    You can still integrate it with the Neptune logger using legacy Python API
+                    available as part of neptune-contrib package:
+                      - https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n
+                    """
+                ) from e
+
+        if not self._run_instance_initialized:
+            # make sure that we've log integration version for both newly created and outside `Run` instances
+            self._run_instance_initialized = True
+            self._run_instance[INTEGRATION_VERSION_KEY] = __version__
+
+        return self._run_instance
 
     @rank_zero_only
-    def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
-        params = self._convert_params(params)
-        params = self._flatten_dict(params)
-        for key, val in params.items():
-            self.experiment.set_property(f"param__{key}", val)
+    def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:  # skipcq: PYL-W0221
+        r"""
+        Log hyper-parameters to the run.
 
-    @rank_zero_only
-    def log_metrics(self, metrics: Dict[str, Union[torch.Tensor, float]], step: Optional[int] = None) -> None:
-        """
-        Log metrics (numeric values) in Neptune experiments.
+        Hyperparams will be logged under the "<prefix>/hyperparams" namespace.
+
+        **Note**
+
+        You can also log parameters by directly using the logger instance:
+        ``neptune_logger.experiment["model/hyper-parameters"] = params_dict``.
+
+        In this way you can keep hierarchical structure of the parameters.
```

Comment:
```suggestion
        Note:

            You can also log parameters by directly using the logger instance:
            ``neptune_logger.experiment["model/hyper-parameters"] = params_dict``.

            In this way you can keep hierarchical structure of the parameters.
```
for google-style sphinx formatting we use the predefined sections